### PR TITLE
Added tests for command caching

### PR DESCRIPTION
**Pull request by shoatman**

### CODE REVIEW

File: msal/src/test/java/com/microsoft/identity/client/robolectric/tests/mocked/CommandResultCachingTest.java [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+
+        return cacheRecord;
+    }
+
+    private IAccount loadAccountForTest(IPublicClientApplication application) {
+        ICacheRecord cacheRecord = createDataInCache(application);
+        final String loginHint = cacheRecord.getAccount().getUsername();
+        final IAccount account = performGetAccount(application, loginHint);
+        return account;
+    }
```
Comment (shahzaibj) [inline msal/src/test/java/com/microsoft/identity/client/robolectric/tests/mocked/CommandResultCachingTest.java]: Are these methods needed? They are not being used anywhere.

File: msal/src/test/java/com/microsoft/identity/client/robolectric/utils/AcquireTokenTestHelper.java [hunk 2]
```diff
@@ -67,6 +69,28 @@ public void onCancel() {
         return callback;
     }
 
+    public static AuthenticationCallback failedSilentRequestDuplicateCommand() {
```
Comment (shahzaibj) [inline msal/src/test/java/com/microsoft/identity/client/robolectric/utils/AcquireTokenTestHelper.java]: nit: Rename to `failedSilentRequestDuplicateCommandCallback`

File: msal/src/test/java/com/microsoft/identity/client/robolectric/tests/mocked/CommandResultCachingTest.java (hunk 1 above)
Comment (shoatman) [reply msal/src/test/java/com/microsoft/identity/client/robolectric/tests/mocked/CommandResultCachingTest.java]: Yep.  Leftover from when creating the tests.  Deleted.

File: msal/src/test/java/com/microsoft/identity/client/robolectric/utils/AcquireTokenTestHelper.java (hunk 2 above)
Comment (shoatman) [reply msal/src/test/java/com/microsoft/identity/client/robolectric/utils/AcquireTokenTestHelper.java]: fixed.

**Comment by shoatman** [comment]
Fixed outstanding comments.

**Pull request merged**